core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -2,36 +2,106 @@
 /*
  * Copyright 2019 Pengutronix
  * All rights reserved.
+ * Copyright 2019 NXP
  *
  * Rouven Czerwinski <entwicklung@pengutronix.de>
  */
 
 #include <drivers/tzc380.h>
 #include <imx-regs.h>
 #include <imx.h>
+#include <kernel/panic.h>
 #include <mm/core_memprot.h>
 #include <mm/generic_ram_layout.h>
 
+#if (defined(PLATFORM_FLAVOR_mx6solosabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6solosabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6sxsabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6sxsabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6ulevk) \
+	|| defined(PLATFORM_FLAVOR_mx6ul9x9evk) \
+	|| defined(PLATFORM_FLAVOR_mx6ullevk) \
+	|| defined(PLATFORM_FLAVOR_mx7dsabresd))
 void imx_configure_tzasc(void)
 {
+	uint8_t region = 1;
+	vaddr_t addr;
 
+	addr = core_mmu_get_va(TZASC_BASE, MEM_AREA_IO_SEC);
+
+	if (!addr)
+		panic("Cannot get TZASC base address");
+
+	tzc_init(addr);
+
+	region = tzc_auto_configure(CFG_DRAM_BASE, CFG_DDR_SIZE,
+					TZC_ATTR_SP_NS_RW, region);
+	region = tzc_auto_configure(CFG_TZDRAM_START, CFG_TZDRAM_SIZE,
+					TZC_ATTR_SP_S_RW, region);
+	region = tzc_auto_configure(CFG_SHMEM_START, CFG_SHMEM_SIZE,
+					TZC_ATTR_SP_ALL, region);
+
+	DMSG("Action register: %xl", tzc_get_action());
+}
+
+#elif (defined(PLATFORM_FLAVOR_mx6qsabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6qsabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6dlsabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6dlsabreauto))
+void imx_configure_tzasc(void)
+{
 	int i;
-	vaddr_t addr[2] = {0};
+	vaddr_t addr[2] = { 0 };
 
 	addr[0] = core_mmu_get_va(TZASC_BASE, MEM_AREA_IO_SEC);
 	addr[1] = core_mmu_get_va(TZASC2_BASE, MEM_AREA_IO_SEC);
 
-	for (i = 0; i < 2; i++) {
+	if (!(addr[0] & addr[1]))
+		panic("Cannot get TZASC base address");
```

Comment:
this looks strange (`if (!(addr[0] & addr[1]))`). please crosscheck it's what you expect.

Reply:
Hi @etienne-lms 
Addressed in a6bca6b924766606e377fc187444f92ce92122dc